src/System.Text.Json/tests/Serialization/CustomConverterTests.BadConverters.cs

```diff
@@ -57,27 +58,78 @@ public static void ContraVariantConverterFail()
         }
 
-        private class CanConvertNullConverterAttribute : JsonConverterAttribute
+        private class InvalidConverterAttribute : JsonConverterAttribute
         {
-            public CanConvertNullConverterAttribute() : base(typeof(int)) { }
-
-            public override JsonConverter CreateConverter(Type typeToConvert)
-            {
-                return null;
-            }
+            // converterType is not valid since typeof(int) is not a type that derives from JsonConverter.
+            public InvalidConverterAttribute() : base(converterType: typeof(int)) { }
         }
 
-        private class PocoWithNullConverter
+        private class PocoWithInvalidConverter
         {
-            [CanConvertNullConverter]
+            [InvalidConverter]
             public int MyInt { get; set; }
         }
 
         [Fact]
         public static void AttributeCreateConverterFail()
         {
-            Assert.Throws<InvalidOperationException>(() => JsonSerializer.Serialize(new PocoWithNullConverter()));
-            Assert.Throws<InvalidOperationException>(() => JsonSerializer.Deserialize<PocoWithNullConverter>("{}"));
+            InvalidOperationException ex;
+
+            ex = Assert.Throws<InvalidOperationException>(() => JsonSerializer.Serialize(new PocoWithInvalidConverter()));
+            // Message should be in the form "The converter specified on 'System.Text.Json.Serialization.Tests.CustomConverterTests+PocoWithInvalidConverter.MyInt' does not derive from JsonConverter or have a public parameterless constructor."
+            Assert.Contains("'System.Text.Json.Serialization.Tests.CustomConverterTests+PocoWithInvalidConverter.MyInt'", ex.Message);
+
+            ex = Assert.Throws<InvalidOperationException>(() => JsonSerializer.Deserialize<PocoWithInvalidConverter>("{}"));
+            Assert.Contains("'System.Text.Json.Serialization.Tests.CustomConverterTests+PocoWithInvalidConverter.MyInt'", ex.Message);
+        }
+
+        private class InvalidTypeConverterClass
+        {
+            [JsonConverter(typeof(JsonStringEnumConverter))]
+            public ICollection<InvalidTypeConverterEnum> MyEnumValues { get; set; }
+        }
+
+        private enum InvalidTypeConverterEnum
+        {
+            Value1,
+            Value2,
+        }
+
+        [Fact]
+        public static void AttributeOnPropertyFail()
+        {
+            InvalidOperationException ex;
+
+            ex = Assert.Throws<InvalidOperationException>(() => JsonSerializer.Serialize(new InvalidTypeConverterClass()));
+            // Message should be in the form "The converter specified on 'System.Text.Json.Serialization.Tests.CustomConverterTests+InvalidTypeConverterClass.MyEnumValues' is not compatible with the type 'System.Collections.Generic.ICollection`1[System.Text.Json.Serialization.Tests.ExceptionTests+InvalidTypeConverterEnum]'."
```

Comment:
This comment is still not correct. We don't have `**ExceptionTests**+InvalidTypeConverterEnum`.

Shouldn't this be:
`System.Text.Json.Serialization.Tests.CustomConverterTests+InvalidTypeConverterEnum`